recipe-client-addon/lib/NormandyDriver.jsm

```diff
@@ -156,10 +156,37 @@ this.NormandyDriver = function(sandboxManager) {
       sandboxManager.removeHold(`setTimeout-${token}`);
     },
 
-    /*
-     * Return a promise that resolves to an Addon ID if installation is successful.
-     */
-    installAddon(url) {
+    addons: {
+      /**
+       * Get information about an installed add-on by ID.
+       *
+       * @param {string} addonId
+       * @returns {Object} Object containing id, installDate,
+       *                   isActive, name, type, and version of addon.
+       */
+      get(addonId) {
+        return new sandbox.Promise((resolve, reject) => {
+          AddonManager.getAddonByID(addonId, addon => {
```

Comment:
`getAddonByID` actually also returns a promise, so you can make this whole function async.